docs/basic/merge-sort.md

```diff
@@ -29,24 +75,52 @@ void merge(int ll, int rr) {
   }
   for (int i = ll; i < rr; ++i) a[i] = t[i];
 }
+// The key point of this code block is that only create array once to avoid useless construction and destruction of object in every recursive call.
 ```
 
-关键点在于一次性创建数组，避免在每次递归调用时创建，避免了对象的无谓构造和析构。
-
-## 逆序对
+### Python
+
+```python
+# Python Version
+def merge_sort(ll, rr):
+    if rr - ll <= 1:
+        return
+    mid = math.floor((rr + ll) / 2)
+    merge_sort(ll, mid)
+    merge_sort(mid, rr)
+    p = s = ll
+    q = mid
+    while(s < rr):
+        if p >= mid or (q < rr and a[p] > a[q]):
+            s += 1
+            q += 1
+            t[s] = a[q]
+        else:
+            s += 1
+            p += 1
+            t[s] = a[p]
+    for i in range(ll, rr):
+        a[i] = t[i]
+```
 
-归并排序还可以用来求逆序对的个数。
+## Inversion
 
-所谓逆序对，就是满足 $a_{i} > a_{j}$ 且 $i < j$ 的数对 $(i, j)$ 。
+Merge sort can be used to find the [inversion number](https://en.wikipedia.org/wiki/Inversion_(discrete_mathematics)#Inversion_number).
 
-可以用 [树状数组](../ds/fenwick.md) 、 [线段树](../ds/seg.md) 等数据结构来求，也可以用归并排序来求。
+A inversion pair $(i,j)$ is a number pair that satisfied $a_{i} > a_{j}$ and $i < j$.
 
-具体来说，上面归并排序中间注释掉的 `ans += mid - p` 就是在统计逆序对个数。
+<!---
+From Wikipedia:
+> Let $\pi$ be a permutation. If $ i < j $ and $\pi(i)>\pi(j)$, either the pair of places $(i,j)$ or the pair of elements $\bigl (\pi (i),\pi (j)\bigr )$ is called an inversion of $\pi$. 
+-->
+<!---原文过于Ambiguous了-->
 
-是因为，那里把靠后的数放到前面了（较小的数放在前面），所以在这个数的原来位置之前的、比它大的数都会和它形成逆序对，而这个个数就是还没有合并进去的数的个数，即为 `mid - p` 。
+The commented statement `ans += mid - p` in previous code implementation is counting the inversion number. Specifically, the algorithm replaces latter elements to former places, i.e, smaller<!---less? 不太对--> elements are placed former. Therefore, any other element former than the original position and greater than the element becomes inversion pairs with this element. The inversion number is the number of elements not yet combined, `mid - p`.
```

Comment:
```suggestion
The commented statement `ans += mid - p` in previous code implementation is counting the inversion number. Specifically, when the algorithm places latter elements to former places, (since smaller elements are placed former), any other element former than the original position and greater than the element forms inversion pairs with this element. The inversion number is the number of elements not yet combined, `mid - p`.
```

Reply:
啊学到了（？

Reply:
不过这里要另外改一下